examples/function_as_tool.py

```diff
@@ -22,20 +22,19 @@ def get_current_weather(
     return f"{location} is 65 degrees {unit}."
 
 
-class CurrentWeatherPrompt(BasePrompt):
+class CurrentWeatherPrompt(OpenAIPrompt):
     """What's the weather like in San Francisco, Tokyo, and Paris?"""
 
     call_params = OpenAICallParams(
         model="gpt-3.5-turbo-1106", tools=[get_current_weather]
     )
 
 
-chat = OpenAIChat()
-completion = chat.create(CurrentWeatherPrompt())
+completion = CurrentWeatherPrompt().create()
 
 if tools := completion.tools:
     for tool in tools:
         if tool.fn:
-            print(tool.fn(**tool.__dict__))
+            print(tool.fn(**tool.model_dump(exclude={"tool_call"})))
```

Comment:
this will have merge conflicts with my recent updates. Please resolve. This should use `tool.args`

Reply:
resolved